src/Microsoft.DotNet.Build.Tasks/PackageFiles/CloudTest.targets

```diff
@@ -0,0 +1,234 @@
+    <Error Condition="'$(BuildCompleteConnection)' == ''" Text="Missing required property BuildCompleteConnection." />
+    <Error Condition="'$(BuildIsOfficialConnection)' == ''" Text="Missing required property BuildIsOfficialConnection." />
+    <!-- verify the test archives were created -->
+    <Error Condition="'@(ForUpload->Count())' == '0'" Text="Didn't find any test archives in '$(ArchivesRoot)'." />
+    <!-- add relative blob path metadata -->
+    <ItemGroup>
+      <ForUpload>
+        <RelativeBlobPath>$(Platform)$(ConfigurationGroup)\Tests\%(Filename)%(Extension)</RelativeBlobPath>
+      </ForUpload>
+    </ItemGroup>
+  </Target>
+
+  <!-- create Azure containers and file shares -->
+  <Target Name="CreateAzureStorage">
+    <CreateAzureContainer
+      AccountKey="$(DropAccessToken)"
+      AccountName="$(AzureContainerAccountName)"
+      ContainerName="$(ContainerName)"
+      ReadOnlyTokenDaysValid="30">
+        <Output TaskParameter="StorageUri" PropertyName="DropUriRoot" />
+        <Output TaskParameter="ReadOnlyToken" PropertyName="DropUriReadOnlyToken" />
+    </CreateAzureContainer>
+    <!-- append the build arch and type to the root URI -->
+    <CreateProperty Value="$(DropUriRoot)/$(BuildArch)$(BuildType)">
+      <Output TaskParameter="Value" PropertyName="DropUri" />
+    </CreateProperty>
+    <CreateAzureFileShare
+      AccountKey="$(FileShareAccessToken)"
+      AccountName="$(AzureFileShareAccountName)"
+      ShareName="results"
+      ReadOnlyTokenDaysValid="30"
+      WriteOnlyTokenDaysValid="1">
+        <Output TaskParameter="StorageUri" PropertyName="ResultsUri" />
+        <Output TaskParameter="ReadOnlyToken" PropertyName="ResultsReadOnlyToken" />
+        <Output TaskParameter="WriteOnlyToken" PropertyName="ResultsWriteOnlyToken" />
+    </CreateAzureFileShare>
+  </Target>
+
+  <Target Name="CreateTestListJson"
+          DependsOnTargets="CreateFuncTestListJson;CreatePerfTestListJson" />
+
+  <Target Name="CreateFuncTestListJson" DependsOnTargets="CreateAzureStorage">
+    <!-- create item group of functional tests -->
+    <ItemGroup>
+      <FunctionalTest Include="$(TestArchivesRoot)*.zip" />
+    </ItemGroup>
+    <ItemGroup>
+      <FunctionalTest>
+        <Command>%HELIX_PYTHONPATH% %HELIX_SCRIPT_ROOT%\xunitrunner.py --dll %(Filename).dll</Command>
+        <CorrelationPayloadUris>[$(DropUri)$(Platform)$(ConfigurationGroup)/Packages.zip$(DropUriReadOnlyToken);;http://dotnetbuildscripts.blob.core.windows.net/scripts/xunit_210.zip;;http://dotnetbuildscripts.blob.core.windows.net/scripts/amd64ret/corerun.zip]</CorrelationPayloadUris>
+        <PayloadUri>$(DropUri)$(Platform)$(ConfigurationGroup)/Tests/%(Filename)%(Extension)$(DropUriReadOnlyToken)</PayloadUri>
+        <WorkItemId>FunctionalTest.%(Filename)</WorkItemId>
+      </FunctionalTest>
+    </ItemGroup> 
+    <WriteItemsToJson JsonFileName="$(FuncTestListFile)" Items="@(FunctionalTest)" />
```

Comment:
This isn't how the ToF builds do it, as it become a real nightmare to maintain this way. (For example, this doesn't work for linux or mac, because of the use of % and \.  We had a template system in ToF that tried to reduce all the duplication of data here... can we not port that?